.github/workflows/sonarcloud.yml

```
@@ -0,0 +1,80 @@
+
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@ed408507eac070d1f99cc633dbcf757c94c7933a # v4.4.3
+      with:
+        validate-wrappers: true
+        cache-read-only: ${{ github.ref != 'refs/heads/main' }}
+
+    - name: Run tests with coverage and quality analysis
+      run: ./gradlew --no-daemon testWithCoverage detekt
+
+    - name: SonarCloud Scan
+      if: ${{ secrets.SONAR_TOKEN != '' }}
```

actionlint (1.7.7)

context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)